### ISSUE TITLE
Malleability Attack

### ISSUE THREAD

**Issue by neocarmack** [omnilaboratory/obd#10]
Bitcoin raw transactions are malleable, if without SegWit. That is to say a transaction can be modified, without invalidate it, but without access to the relevant private keys.
 
[BIP-0062](https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki#motivation)

So that OBD HTLC transactions built upon omni raw transactions are maleable, one participant ( Alice ) can broadcast a modified elder commitment transaction with totally different TxID without trigger the following ED, BR, HERD transactions, but pays himself the same money. 

His counterparty (Bob) has no knowledge of this actitvity, and even if Bob knows, his punishment BR transaction can never be broadcast because Alice broadcaseted an transaction with totally different TxID.  

When Omnicore supports SegWit, obd shall update all the currently being constructed transacitons.

**Comment by neocarmack** [comment]
[Segregated Witness Benefits](https://bitcoincore.org/en/2016/01/26/segwit-benefits/)

[Bitcoin Magazine article on 2015 Malleability attack](https://bitcoinmagazine.com/articles/the-who-what-why-and-how-of-the-ongoing-transaction-malleability-attack-1444253640)

[BIP-62](https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki#motivation)

[BIP 140 – alternative approach to malleability fixes](https://github.com/bitcoin/bips/blob/master/bip-0140.mediawiki)

[Transaction Malleability in the blockchain](https://bitcoin.stackexchange.com/questions/22051/transaction-malleability-in-the-blockchain/22058#22058)

**Comment by DeusNexus** [comment]
Is this Malleability attack fixed?

**Comment by neocarmack** [comment]
yes. But i will close this issue after omnibolt mainnet launches. Currently we tested on testnet.
